tfhe/src/shortint/key_switching_key/mod.rs

```diff
@@ -250,9 +378,119 @@ impl KeySwitchingKey {
     ///
     /// assert_eq!(ck2.decrypt(&cipher_2), cleartext);
     /// ```
-    pub fn cast(&self, ct: &Ciphertext) -> Ciphertext {
-        let mut ret = self.dest_server_key.create_trivial(0);
-        self.cast_into(ct, &mut ret);
+    pub fn cast(&self, input_ct: &Ciphertext) -> Ciphertext {
+        let output_lwe_size = match self.destination_key {
+            EncryptionKeyChoice::Big => self
+                .dest_server_key
+                .bootstrapping_key
+                .output_lwe_dimension()
+                .to_lwe_size(),
+            EncryptionKeyChoice::Small => self
+                .dest_server_key
+                .bootstrapping_key
+                .input_lwe_dimension()
+                .to_lwe_size(),
+        };
+        let mut keyswitched = self
+            .dest_server_key
+            .unchecked_create_trivial_with_lwe_size(0, output_lwe_size);
+
+        match self.cast_rshift {
```

Comment:
nitpick, to avoid the _ => unreachble, maybe we could do

```rust
match self.cast_rshift.cmp(0) {
 Ordering::Eq => {}
 Ordering::Less => {}
 Ordering::Greater => {}
 
}
```

Reply:
was from the previous code that's a good point